brozzler/worker.py

```diff
@@ -242,53 +242,98 @@ def brozzle_page(
+                except brozzler.ReachedLimit as e:
+                    raise
+                except brozzler.ShutdownRequested:
+                    raise
+                except brozzler.ProxyError:
+                    raise
+                except Exception as e:
+                    if (
+                        hasattr(e, "exc_info")
+                        and len(e.exc_info) >= 2
+                        and hasattr(e.exc_info[1], "code")
+                        and e.exc_info[1].code == 430
+                    ):
+                        self.logger.info(
+                            "youtube-dl got %s %s processing %s",
+                            e.exc_info[1].code,
+                            e.exc_info[1].msg,
+                            page.url,
+                        )
+                    else:
+                        self.logger.error(
+                            "youtube_dl raised exception on %s", page, exc_info=True
+                        )
+        else:
+            self.logger.info("needs fetch: %s", page)
+            self._fetch_url(site, page=page)
         return outlinks
 
+    def _get_page_headers(self, page):
+        with requests.get(page.url, stream=True) as r:
+            content_type_header = content_length_header = last_modified_header = None
+            if "Content-Type" in r.headers:
+                content_type_header = "Content-Type"
+            elif "content-length" in r.headers:
+                content_type_header = "content-length"
+            elif "CONTENT-LENGTH" in r.headers:
+                content_type_header = "CONTENT-LENGTH"
+            if content_type_header:
+                page.content_type = r.headers[content_type_header]
+                self.logger.info(
+                    "url %s content_type is %s", page.url, page.content_type
+                )
+
+            if "Content-Length" in r.headers:
+                content_length_header = "Content-Length"
+            elif "content-length" in r.headers:
+                content_length_header = "content-length"
+            elif "CONTENT-LENGTH" in r.headers:
+                content_length_header = "CONTENT-LENGTH"
+            if content_length_header:
+                page.content_length = int(r.headers[content_length_header])
+                self.logger.info(
+                    "url %s content_length is %s", page.url, page.content_length
+                )
+
+            if "Last-Modified" in r.headers:
+                last_modified_header = "Last-Modified"
+            elif "Last-Modified" in r.headers:
+                last_modified_header = "Last-Modified"
+            elif "LAST-MODIFIED" in r.headers:
+                last_modified_header = "LAST-MODIFIED"
+            if last_modified_header:
+                page.last_modified = r.headers[last_modified_header]
+                self.logger.info(
+                    "url %s last_modified is %s", page.url, page.last_modified
+                )
```

Comment:
I have a suggestion to make a code improvement here.
If you use a dict-like data structure that is **case insensitive** like this one
https://github.com/psf/requests/blob/main/src/requests/structures.py#L13
you won't have to write all this code and you'll catch ALL cases.
E.g. maybe you are missing some edge cases like `LAST-modified`, people do weird things on the web :) 
I mention this because you already have `requests` in your dependencies and it would be easy to use it.
An alternative one is this one : https://werkzeug.palletsprojects.com/en/3.0.x/datastructures/#werkzeug.datastructures.Headers

Reply:
Oh, I'm sorry but now that I'm looking it again, I see that you are checking a response from the `request` libraries. Maybe the data structure I mentioned is already used and you can take advantage of it right now. Please try to remove all these alternative upper-case / lower-case combinations and try to use only lower-case to check that it works.

Reply:
yes, thank you, @vbanos!  `requests` uses its `CaseInsensitiveDict` for the headers here, and I've updated.